crates/cheatnet/src/cheatcodes/prank.rs

```diff
@@ -1,17 +1,45 @@
+use crate::state::{CheatStatus, CheatTarget};
 use crate::CheatnetState;
 use starknet_api::core::ContractAddress;
 
 impl CheatnetState {
-    pub fn start_prank(
-        &mut self,
-        contract_address: ContractAddress,
-        caller_address: ContractAddress,
-    ) {
-        self.pranked_contracts
-            .insert(contract_address, caller_address);
+    pub fn start_prank(&mut self, target: CheatTarget, caller_address: ContractAddress) {
+        match target {
+            CheatTarget::All => {
+                self.global_prank = Some(caller_address);
+                // Clear individual pranks so that `All`
+                // contracts are affected by this prank
+                self.pranked_contracts.clear();
+            }
+            CheatTarget::One(contract_address) => {
+                self.pranked_contracts
+                    .insert(contract_address, CheatStatus::Cheated(caller_address));
+            }
+            CheatTarget::Multiple(contract_addresses) => {
+                for contract_address in contract_addresses {
+                    self.pranked_contracts
+                        .insert(contract_address, CheatStatus::Cheated(caller_address));
+                }
+            }
+        }
```

Comment:
I wonder if we can make a general function just like you did 
with `get_cheat_for_contract` (really good idea btw). We then could pass different mutable references to the state for different cheatcodes (e.g. for `prank` we would pass `&mut self.pranked_contracts` and `&mut self.global_prank`).